features/bootstrap/Aws/Test/Integ/S3Context.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Aws\Test\Integ;
+
+use Aws\S3\S3Client;
+use Behat\Behat\Tester\Exception\PendingException;
+use Behat\Behat\Context\Context;
+use Behat\Behat\Context\SnippetAcceptingContext;
+use Behat\Gherkin\Node\PyStringNode;
+use Behat\Gherkin\Node\TableNode;
+use Aws\S3\PostObject;
+use Aws\S3\BatchDelete;
+use Aws\ResultInterface;
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7;
+use PHPUnit_Framework_Assert as Assert;
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Defines application features from the specific context.
+ */
+class S3Context implements Context, SnippetAcceptingContext
+{
+    use IntegUtils;
+
+    const MB = 1048576;
+    const RESOURCE_POSTFIX = 'php-integration-s3-post-object-test';
+
+    /** @var S3Client */
+    private $s3Client;
+    /** @var StreamInterface */
+    private $stream;
+    /** @var ResultInterface */
+    private $result;
+
+    private static $tempFile;
+
+    /**
+     * @Given I have an :S3 client
+     */
+    public function iHaveAnClient()
+    {
+        $this->s3Client = self::getSdk()->createS3();
+        $this->stream = Psr7\stream_for(Psr7\try_fopen(self::$tempFile, 'r'));
+    }
+
+    /**
+     * @When I do a POST upload
+     */
+    public function iDoAPostUpload()
+    {
+        date_default_timezone_set('UTC');
+        $ldt = gmdate(PostObject::ISO8601_BASIC);
+        $sdt = substr($ldt, 0, 8);
+
+        $scope = "$sdt/{$this->s3Client->getRegion()}/s3/aws4_request";
+        $credentails = "{$this->s3Client->getCredentials()->wait()->getAccessKeyId()}/$scope";
+
+        $policy = [
+            'expiration' => gmdate('Y-m-d\TH:i:s\Z', strtotime('+1 hour')),
+            'conditions' => [
+                ["bucket" => self::getResourceName()],
+                ["starts-with", '$key', ""],
+                ["acl" => "public-read"],
+                ["x-amz-credential" => $credentails],
+                ["x-amz-algorithm" => "AWS4-HMAC-SHA256"],
+                ["x-amz-date" => $ldt],
+            ],
+        ];
```

Comment:
Since a V4 policy needs to include conditions about the `x-amz-date`, `x-amz-credential`, and `x-amz-algorithm` fields, I don't believe this is a backwards compatible change. Any policy that is currently valid for a SigV2 POST upload would fail with a message like 'Invalid according to Policy: Extra input fields: x-amz-date' if the PostObject class suddenly started using V4 signatures.